src/main/java/org/jabref/gui/util/CustomFilteredList.java

```diff
@@ -0,0 +1,321 @@
+package org.jabref.gui.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.function.Predicate;
+
+import javafx.beans.NamedArg;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.ObjectPropertyBase;
+import javafx.collections.ListChangeListener.Change;
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.TransformationList;
+
+/**
+ * A custom class that extends {@link javafx.collections.transformation.FilteredList FilteredList} to provide additional functionality.
+ * This class closely mirrors the behavior of {@code FilteredList} with the following key differences:
+ * <ol>
+ * <li>Offers access to the {@link #refilter()} method, to re-evaluation of the filter predicate over the entire list.</li>
```

Comment:
I would like to hear about the "why"

It seems this is necessary, because there is no update going through the `source` list.

I would love to see a MWE. Meaning: A minimal Java project showing the issue. I assume, EasyBind needs to be involed in that setting?

Also OK as future work. I just want to get rid of this class in the next weeks.

Reply:
1. I want the `FilteredList` to have only one predicate function, as it needs to be fast since it's running on the UI thread. The predicate will only check for a boolean value in the `BibEntryTableViewModel`.
In the background thread, after the search query/selected groups changes, the matching function will be executed and the boolean value in `BibEntryTableViewModel` will be updated. After updating, I need to refilter the list (which is a private method in JavaFX), so I need the refilter method to be public.

2. When updating/adding an entry, updating matchs should be called to update the boolean value in the background. After updating, I need to test the predicate function again for this entry, so I created a refilter method that accepts an index parameter.